### PR TITLE
feat: add label-on-comment workflow from BTR WG

### PR DESCRIPTION
**Pull request by Carlos-Muniz**
The `label-on-comment` workflow is going to be used in other repos other than just the BTR WG repo. In order to make this more standard, this workflow is being moved to this repo, it's name changed to `add-remove-label-on-comment` and will be then inherited by workflows in other repos, such as the BTR WG repo.

### CODE REVIEW

**Review by sarina — CHANGES_REQUESTED** [review]
You should be adding a workflow template too, see https://github.com/openedx/.github/tree/master/workflow-templates

File: workflow-templates/label-on-comment.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# label "really long lots of words label"
+
+name: Allows for the adding and removing of labels via comment
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  add_label:
```
Comment (sarina) [inline workflow-templates/label-on-comment.yml:18]: I would call this job "process_comment" or "add_delete_labels" or something. 

Here's the way this works:

- You're defining ONE job here that references ALL the jobs in the file below
- If that file contains multiple jobs, all will be run

Thus, the job you define to _call_ this file should be descriptive of all jobs that are being run.

Based on this comment I actually think we should change the name of the workflow to `add-remove-label-on-comment`. I named this file when only having the "add" action, now it also has the "delete" action.

Comment (Carlos-Muniz) [reply workflow-templates/label-on-comment.yml]: Thank you for explaining this. I've updated all the workflow template with the job name `add_remove_labels`, changed the name of the workflow template `add-remove-label-on-comment.yml`, and updated the workflows in BTR WG and public engineering repos.

Comment (sarina) [reply workflow-templates/label-on-comment.yml]: One last thing, you also need to rename the source workflow too.

Comment (Carlos-Muniz) [reply workflow-templates/label-on-comment.yml]: Done!

**Pull request merged**